coderd/httpapi/httpapi.go

```diff
@@ -232,16 +243,22 @@ func ServerSentEventSender(rw http.ResponseWriter, r *http.Request) (func(ctx co
 			return err
 		}
 
-		mu.Lock()
-		defer mu.Unlock()
-		_, err = rw.Write(buf.Bytes())
-		if err != nil {
-			return err
+		event := sseEvent{
+			payload: buf.Bytes(),
+			errC:    make(chan error, 1),
 		}
-		f.Flush()
 
-		return nil
+		select {
+		case <-r.Context().Done():
+			return r.Context().Err()
+		case <-ctx.Done():
+			return ctx.Err()
+		case eventC <- event:
+			return <-event.errC
+		case <-closed:
+			return xerrors.New("server sent event sender closed")
```

Comment:
Should be before the write to the events channel

Reply:
Why? Order doesn't matter for `select`.

Reply:
Yeah I wrote this comment before I remembered it's an unbuffered channel so it's not relevant

Reply:
Ah, gotcha. Btw, to go runtime intentionally randomizes which channel is picked, so even in the event of a buffered channel, the order wouldn't matter as we can't control which available channel is picked.